### PR TITLE
Docs: Data Model

### PR DESCRIPTION
**Pull request by erondpowell**
## Description
Updates to the Data Models Docs

### TODO
- [x] Lay out new initial page structure
- [x] Redistribute content from old Data Model pages into new pages
- [x] "Hide" the Fields pages (to be released in a second PR)
- [x] Draft Data Model
- [x] Draft Collections
- [x] Draft System Collections
- [x] Draft Fields
- [x] Draft Relations
- [x] Add Sketches https://github.com/directus/docs/issues/51
- [x] Add Media
- [x] Final Proofedits
- [x] Tag manager for editorial review

### CODE REVIEW

File: configuration/data-model/relationships.md [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+
+When we describe the relationship between two tables, remember that the relationships are always described relative to their "parent" collection. However, this naming convention is based on perspective. For example, within the data model, a [many-to-one relationship](#many-to-one-m2o) is the same as a [one-to-many relationship](#one-to-many-o2m), the term used just depends on which collection you consider to be the parent.
+
+## Many-to-One (M2O)
+
+<video title="Configure an M2O Relationship" autoplay muted loop controls>
+	<source src="" type="video/mp4" />
+</video>
+
+The easiest way to configure an M2O field is to follow the guide on how to [create a field (standard)](/configruation/data-model/fields/#create-a-field-standard) and select the M2O Interface from the template wizard.
```
Comment (rijkvanzanten) [inline configuration/data-model/relationships.md:45]: ~configruation~ configuration

**Comment by azrikahar** [comment]
> In general, make sure to run a spellchecker.

Maybe we can also explore [spellchecker-cli](https://www.npmjs.com/package/spellchecker-cli) as a git hook or github action. 🤔

It was what I used when I made the PR https://github.com/directus/directus/pull/9553. It may be non-trivial to setup a dictionary at first (to "whitelist" words such as `Directus`, `O2M`, `M2A` etc), however the bright side is that'll come with the advantage of standardizing the letter casing for words/terms that have been discussed for a while.

**Comment by rijkvanzanten** [comment]
### Relationships

> Therefore, aside from any technical limitations of your project's infrastructure or core requirements for any relational data model, like having a primary key field for every collection or a data type assigned to every field, you are free to build the data model as you want, **for better or worse**.

(emphasis mine) Little too ominous IMO. I think we can drop this whole sentence.

> Therefore, in order to build an appropriate, efficient, and effective relational data model, you still need a rock solid conceptual understanding of relational data models.

I don't think you need a _rock solid_ conceptual understanding in order to build a data model. It feels a little too gate-keep-y to me. You can use Directus with a minimal understanding, but really utilize all it's powers when you have a better understanding.

**Comment by erondpowell** [comment]
> Lets flip the "The easiest way to configure a ..." with the explanation. We should explain first, and then end with the CTA of "And here's how to configure it"

My thought was that a good number of people will already know how that works, and are hunting for how to do it in the app. This would save them time. But I will ditch that thought and flip it anyways.

**Comment by erondpowell** [comment]
> > In general, make sure to run a spellchecker.
> 
> Maybe we can also explore [spellchecker-cli](https://www.npmjs.com/package/spellchecker-cli) as a git hook or github action. 🤔
> 
> It was what I used when I made the PR [directus/directus#9553](https://github.com/directus/directus/pull/9553). It may be non-trivial to setup a dictionary at first (to "whitelist" words such as `Directus`, `O2M`, `M2A` etc), however the bright side is that'll come with the advantage of standardizing the letter casing for words/terms that have been discussed for a while.

I also talked to @rijkvanzanten about putting the glossary into like YAML or JSON, then instead of repeating very similar explanations throughout the docs, we inject them with a special Vue component. I think he was fairly receptive? Maybe we could combine both efforts, as they are somewhat similar in nature.

**Comment by rijkvanzanten** [comment]
Very receptive to the shared glossary idea, but I do think it's very different from setting up continuous integration automation to check spelling on PRs, so lets keep those as separate efforts 🙂

**Comment by erondpowell** [comment]
@rijkvanzanten 

> The last half isn't necessarily true. It really depends on the context of where the values are used. In most programming languages, it will crash as you're mixing types. It's only in weakly typed programming languages like javascript where you can get away with things like 2 + '2'.

I was trying to highlight addition vs string concatenation, not addition & type mixing.
Should I clarify the message with `"2" + "2"` and a little more verbal explanation?
Will that work? or am i still missing something?

**Comment by erondpowell** [comment]
> Therefore, setting the right data type on each column is an extremely important step in data model configuration.

>> This is becoming less and less important though. Initially, it was mostly to save on storage space and increase database performance, but nowadays storage is cheap enough, and computers fast enough, where stuff is increasingly just stored unstructured

mmm, yeah. I see. I was really trying to highlight the importance of knowing what kind of data type you are working with, so when you feed it into a function, you don't get wacky results. Is that not a real concern?

**Comment by erondpowell** [comment]
Keys and IDs
Lets add a short description to each of the primary key types (eg what's the difference between "auto inc integer vs auto inc big integer"

Adding it for now, but I think we should put that level of detail deeper into the documentation, like in the `fields > field config drawer docs` ... or the `glossary`.

**Pull request merged**